package step2;

import java.io.*;

public class CatchException {
    public static void main(String[] args) {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        String str;
        /********** Begin *********/
        //补全try/catch块，异常类型为IOException
        int letter=0,space=0,number=0,other=0,i=0;
        try {
            str = reader.readLine();
            
            for(i=0;i<str.length();i++){
                if(Character.isLetter(str.charAt(i))){
                    letter++;}
                else if(Character.isSpaceChar(str.charAt(i))){
                    space++;}
                else if(Character.isDigit(str.charAt(i))){
                    number++;} 
                else{
                    other++;
                }
           }
        }
        catch ( Exception e ){
            System.out.println("\t产生的异常已被处理，该catch块不会被执行。"); 
        }

        System.out.printf("letter="+letter+",space="+space+",number="+number+",other="+other);
        /********** End *********/
    }
}